### PR TITLE
KAA-1011 Fix cppcheck integration

### PR DESCRIPTION
**Pull request by rasendubi**

### CODE REVIEW

**Comment by RostakaGmfun** [comment]
Run build

File: client/client-multi/client-c/src/kaa/kaa_notification_manager.c [hunk 1]
```diff
@@ -1318,10 +1314,8 @@ kaa_error_t kaa_notification_manager_handle_server_sync(kaa_notification_manager
                 KAA_LOG_INFO(self->logger, KAA_ERR_NONE, "Received topics list. Topics count is %u", topic_count);
                 shift_and_sub_extension(reader, &extension_length, sizeof(uint16_t));
 
-                kaa_topic_t *topic = NULL;
-
                 while (topic_count--) {
-                   topic = (kaa_topic_t *) KAA_MALLOC(sizeof(kaa_topic_t));
+                    kaa_topic_t *topic = (kaa_topic_t *) KAA_MALLOC(sizeof(kaa_topic_t));
```
Comment (forGGe) [inline client/client-multi/client-c/src/kaa/kaa_notification_manager.c]: odd whitspace

File: client/client-multi/client-c/src/kaa/kaa_notification_manager.c [hunk 2]
```diff
@@ -1318,10 +1314,8 @@ kaa_error_t kaa_notification_manager_handle_server_sync(kaa_notification_manager
...
                    KAA_RETURN_IF_NIL(topic, KAA_ERR_NOMEM);
```
Comment (forGGe) [inline client/client-multi/client-c/src/kaa/kaa_notification_manager.c]: Whitespace

File: client/client-multi/client-c/test/collections/test_kaa_list.c [hunk 3]
```diff
@@ -104,10 +104,9 @@ static void test_list_sort()
     kaa_list_t *list = kaa_list_create();
     ASSERT_NOT_NULL(list);
 
-    test_list_node_t *node;
     uint64_t node_number = 100;
     for (uint64_t i = 0; i < node_number; ++i) {
-        node = (test_list_node_t *)KAA_MALLOC(sizeof(test_list_node_t));
+        test_list_node_t *node = (test_list_node_t *)KAA_MALLOC(sizeof(test_list_node_t));
```
Comment (forGGe) [inline client/client-multi/client-c/test/collections/test_kaa_list.c]: Redundant cast

**Comment by forGGe** [comment]
Minor nitpicks, looks good otherwise

**Comment by forGGe** [comment]
Looks good

**Pull request merged**